wpilib/wpilib/joystick.py

```diff
@@ -171,76 +287,59 @@ def getAxis(self, axis):
         elif axis == self.AxisType.kThrottle:
             return self.getThrottle()
         else:
-            raise ValueError("Invalid axis specified! Must be one of wpilib.Joystick.AxisType, or use getRawAxis instead")
-        
-    def getAxisCount(self):
-        """For the current joystick, return the number of axis"""
-        return self.ds.getStickAxisCount(self.getPort())
+            raise ValueError(
+                "Invalid axis specified! Must be one of wpilib.Joystick.AxisType, or use getRawAxis instead")
 
-    def getTrigger(self, hand=None):
+    def getTrigger(self):
         """Read the state of the trigger on the joystick.
 
         Look up which button has been assigned to the trigger and read its
         state.
 
-        :param hand: This parameter is ignored for the Joystick class and is
-            only here to complete the GenericHID interface.
         :returns: The state of the trigger.
         :rtype: bool
         """
-        return self.getRawButton(self.buttons[self.ButtonType.kTrigger])
+        return self.getRawButton(self.Button.kTrigger)
 
-    def getTop(self, hand=None):
-        """Read the state of the top button on the joystick.
-
-        Look up which button has been assigned to the top and read its state.
+    def getTriggerPressed(self):
+        """Whether the trigger was pressed since the last check
 
-        :param hand: This parameter is ignored for the Joystick class and is
-            only here to complete the GenericHID interface.
-        :returns: The state of the top button.
-        :rtype: bool
+        :returns: Whether the button was pressed since the last check
         """
-        return self.getRawButton(self.buttons[self.ButtonType.kTop])
+        return self.getRawButton(self.Button.kTrigger)
```

Comment:
getRawButtonPressed instead?

Reply:
Damn my find/replace didnt work. Dont be surprised if you find a bunch of these